starlette/_utils.py

```diff
@@ -10,3 +17,53 @@ def is_async_callable(obj: typing.Any) -> bool:
     return asyncio.iscoroutinefunction(obj) or (
         callable(obj) and asyncio.iscoroutinefunction(obj.__call__)
     )
+
+
+T_co = typing.TypeVar("T_co", covariant=True)
+
+
+# TODO: once 3.8 is the minimum supported version (27 Jun 2023)
+# this can just become
+# class AwaitableOrContextManager(typing.Awaitable[T_co], typing.AsyncContextManager[T_co]):    pass
```

Comment:
```suggestion
# class AwaitableOrContextManager(Awaitable[T_co], AsyncContextManager[T_co]):    pass
```

Reply:
But then this is not a protocol, it's a class, and we're not returning an instance of it.

Reply:
I'm just hiding the `typing` module to make the comment shorter, so it passes the linter :thinking:

Reply:
oh sorry didn't see the suggestion was on the comment. I already reformatted it.